core/domain/suggestion_services.py

```diff
@@ -131,15 +213,23 @@ def create_suggestion(
     suggestion_domain_class = (
         suggestion_registry.SUGGESTION_TYPES_TO_DOMAIN_CLASSES[
             suggestion_type])
+    # Here, argument `final_reviewer_id` of suggestion_domain_class method
+    # can only accept string values but here we are providing None value
+    # which causes MyPy to throw `incompatible argument type` error. Thus
+    # to avoid the error, we used ignore here.
     suggestion = suggestion_domain_class(
         thread_id, target_id, target_version_at_submission, status, author_id,
-        None, change, score_category, language_code, False)
+        None, change, score_category, language_code, False)  # type: ignore[arg-type]
     suggestion.validate()
 
+    # Here, argument `final_reviewer_id` of create method can only accept
+    # string values but here we are providing None value which causes MyPy
+    # to throw `incompatible argument type` error. Thus to avoid the error,
+    # we used ignore here.
```

Comment:
ditto above

Reply:
I think we only maintain 2 issues for type checking, one is for removing wrong type-checks from test cases and another is for ndb properties, can you provide more info on which TODO we add here ( or shall I create a new todo here! ).

Reply:
Or maybe why we add ignore here and not modify the type?

Reply:
Ignore and comment is removed from here!!
But ignore cannot be removed from the above case because in the above case `suggestion_domain_class` is of Union type ( `Union[
        Type[SuggestionEditStateContent],
        Type[SuggestionTranslateContent],
        Type[SuggestionAddQuestion]
    ]` ) and out of all these classes only `SuggestionEditStateContent` can accept Optional[str] for language_code and for other classes which cannot accept Optional[str] for language_code, MYPY throwing an error.

Reply:
Ignore still here

Reply:
Before there were 2 ignores, one of them is removed but other ignore (defined on line `223`) cannot be removed because:
 suggestion_domain_class is of Union type ( Union[ Type[SuggestionEditStateContent], Type[SuggestionTranslateContent], Type[SuggestionAddQuestion] ] ) and out of all these classes only SuggestionEditStateContent can accept Optional[str] for language_code and for other classes that cannot accept Optional[str] for language_code, MYPY throwing an error.

Reply:
Ah I see.